TypeCobol/Compiler/Parser/CodeElementBuilder/CobolCodeElementBuilder.cs

```diff
@@ -57,6 +57,17 @@ internal partial class CodeElementBuilder: CodeElementsBaseListener {
                 }
                 if (CobolExpressionsBuilder.storageAreaGroupsCorrespondingImpact != null) {
                     CodeElement.StorageAreaGroupsCorrespondingImpact = CobolExpressionsBuilder.storageAreaGroupsCorrespondingImpact;
+                    if (CodeElement.StorageAreaWrites == null)
+                    {
+                        CodeElement.StorageAreaWrites=new List<ReceivingStorageArea>();
+                    }
+                    if (CodeElement.StorageAreaReads == null)
+                    {
+                        CodeElement.StorageAreaReads=new List<StorageArea>();
+                    }
+
+                    CodeElement.StorageAreaWrites.Add(new ReceivingStorageArea(StorageDataType.Any, CodeElement.StorageAreaGroupsCorrespondingImpact.SendingGroupItem));
```

Comment:
Can you create a property SendingGroupStorageDataType in class GroupCorrespondingImpact?
This property will be set to StorageDataType.Alphanumeric when SendingGroupItem is set.
(All current usage of SendingGroupItem are of type StorageDataType.Alphanumeric).

Then you can replace StorageDataType.Any with SendingGroupStorageDataType .